[IMP] website: introduce `s_reviews_wall` snippet

chgo-odoo opened the pull request:
This PR introduces the `s_reviews_wall` snippet and reviews the `s_blockquote` options management to achieve something similar to `s_card` allowing to bind options on the parent container rather than the snippet itself

| Snippet in use |
|--------|
| <img width="1712" alt="image" src="https://github.com/user-attachments/assets/d7a56d79-e72d-415a-aae4-733f6fe511a1"> | 

task-4264114
part of task-4077427

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/183965.png)](https://mergebot.odoo.com/odoo/odoo/pull/183965)

Brieuc-brd commented on addons/website/views/snippets/s_reviews_wall.xml:
```diff
@@ -0,0 +1,222 @@
+                        </div>
+                    </div>
+                </div>
+                <div data-name="Card" class="col-12 col-lg-4 pt16 pb16">
```
Hey @chgo-odoo, thanks for your work 👍 

Before reviewing this PR, I think it would be nice to use `s_blockquote` instead of `s_card` with custom content.
That way, we'd inherit blockquote's features and stay consistent with the design and features.
It would also be a good opportunity to improve `s_blockquote` by adding `s_rating` in this snippet with a toggle button to show/hide it 👍 

What do you think?

chgo-odoo commented:
Thanks @Brieuc-brd, using the blockquote snippet is indeed a great suggestion as I was trying to achieve such a design without the right tool 😅 

I've used the blockquote snippet for this one, as you suggested, and added the possibility to add a `s_rating` without the default `s_blockquote`. I've also added a `data-vxml` there as I was modifying the default template, it was needed IIRC.

Could you have a look at the visual result please?

Thanks 🙂

Brieuc-brd reviewed (changes requested):
Nice job @chgo-odoo, this snippet looks cool 👍 
I left a few comments below.

Also, at the moment we can drag and drop inner elements between review blocks and next to the author picture, which is not really nice. Could you have a look at it ?
It seems `s_three_column` is similar to your snippet and doesn't have that, maybe it could help. :)
https://www.awesomescreenshot.com/video/33065570?key=011b5eb238de399b7132eca10388c95a

About commit ba804625fecfc03cfe72850ec96855e3ae4e5b38: " to achieve a ~~more~~ layout more oriented on customer reviews."  Too much "more" I guess 😅

chgo-odoo commented:
Hello @stefanorigano 👋 

Since we are implementing the approach you used on `s_card` to `s_blockquote`, could you have a look at commit https://github.com/odoo/odoo/pull/183965/commits/bb3e51b25da872816c4d22abe08b26e59dce92b2 to ensure we didn't miss anything?

Thanks !

stefanorigano commented on addons/website/views/snippets/s_blockquote.xml:
```diff
@@ -20,55 +34,81 @@
-        </div>
+    <!-- Layout -->
+    <div t-att-data-selector="selector" t-att-data-target="target" t-att-data-exclude="exclude">
+        <we-range string="Edge Spacing" data-select-class="p-1|p-2|p-3|p-4|p-5"/>
```
The blockquote currently comes with an enforced `gap-4` class, which works well for the **current** intended use (e.g., large quote cards in a carousel). However, this new snippet introduces use cases for different scenarios. It would be useful to make the gap editable, especially when the review is brief or there’s no review at all.
What do you think?

|`gap-4` | `gap-0` |
|--------|--------|
| <img width="637" alt="image" src="https://github.com/user-attachments/assets/56d8f498-bfcf-4e2d-9dd6-52a79e8335ac"> | <img width="324" alt="image" src="https://github.com/user-attachments/assets/524ae8a1-beed-4849-bea6-b0b48e901186"> |

chgo-odoo replied:
Not sure about this one, the gap looks fine IMO. I think you still have the gap coming from the empty line if your screenshot, which makes it look super large (?)
<img width="465" alt="image" src="https://github.com/user-attachments/assets/6d78421d-0d1f-4fed-8565-2c6626b26b24">

stefanorigano replied:
Ok, as discussed at the office, let's ignore the scenario for now  👍

stefanorigano commented on addons/website/views/snippets/s_blockquote.xml:
```diff
@@ -7,6 +7,20 @@
+            <h4 class="s_rating_title o_default_snippet_text">Quality</h4>
+            <div class="s_rating_icons o_not_editable">
+                <span class="s_rating_active_icons" style="color: var(--warning)">
+                    <i class="fa fa-star"/>
```
add `role="presentation"` attribute (all other occurrences as well)

chgo-odoo commented:
Hello @stefanorigano, thanks for the nice review !

It makes perfectly sense to review the `s_rating` with a similar design to the one in use here, indeed. Just wanted to double-check something about the hiding the title in `s_rating`, it is currently possible with an option in the web editor, I don't know if you were referring to something else?

lebl-odoo commented:
@chgo-odoo I had a quick look to the PR, looks great! Thanks

![image](https://github.com/user-attachments/assets/367320b5-6ee4-49ec-a52f-ecfe0c7114f0)
Just wondering why the snippets is using "custom" icons instead of "stars" ones? I propose to change the default behavior of "stars" if the main reason is because we wanted a "full star" (and not the outline star).

![image](https://github.com/user-attachments/assets/c0a77b74-90d1-4025-96c1-456680df49de)
Changing number of columns will broke the layout. The title should be part of another div.
What about adding some spacing between cards in "col" mode? Moreover, is it really the default value we want?  
![image](https://github.com/user-attachments/assets/cbc7a419-4ced-40ad-bc5b-aead22547a15)

(didn't pay attention to previous comments, sorry if some are redundants)

chgo-odoo commented:
Thanks @lebl-odoo, about the stars, that's indeed related to the fact that the `star` currently means `fa-star-o` instead of `fa-star`,  this should be addressed in the separate task for the `s_rating`.

About the column issue, the col number should be fixed, indeed. For the spacing, the blockquote should be spaced at the moment, I'll double check

chgo-odoo commented:
@stefanorigano now that we have the `s_rating` merged, this one has the expected structure. Perhaps you want to check it one last time before I move it to tech review 😄

chgo-odoo commented:
@stefanorigano small reminder about this PR 👋 

I fixed the small comments you left, if you could have a look to confirm it's ready for tech review.

Thanks 🙏

stefanorigano commented on addons/website/views/snippets/s_reviews_wall.xml:
```diff
@@ -0,0 +1,195 @@
+        <div class="container">
+            <h2>Customers testimonials</h2>
+            <p class="lead">What our customers are saying about us.</p>
+            <div class="row align-items-stretch s_nb_column_fixed">
```
I think we should remove this class, right ? 😉

```suggestion
            <div class="row align-items-stretch">
```

stefanorigano reviewed (changes requested):
Almost good for me, 2 details 😊
Thanks @chgo-odoo !

qsm-odoo commented:
@robodoo delegate=xO-Tx 
@xO-Tx

xO-Tx reviewed (approved):
LGTM, thanks for the work @chgo-odoo & @Brieuc-brd :+1:
@robodoo rebase-ff r+

robodoo commented:
Merge method set to rebase and fast-forward.
